webhook/strict.go

```diff
@@ -0,0 +1,313 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"context"
+	"github.com/knative/pkg/apis"
+	"reflect"
+	"strings"
+)
+
+// strictValidate validates the following rule sets:
+// - On Create, Spec and Status fields that are marked `Deprecated:` are not
+//   allowed to be set.
+// - On Update, Spec and Status fields that are marked `Deprecated:` are not
+//   allowed to be updated, unless deleting.
+//
+// Note: nil values denote absence of `old` (create) or `new` (delete) objects.
+func strictValidate(ctx context.Context, old GenericCRD, new GenericCRD) error {
+	// Check if it is a delete, if it is we don't validate.
+	if new == nil {
```

Comment:
`new` is a bad name, given it shields the builtin.

Reply:
This is matching the rest of webhook's impl.